src/util/pelz_loaders.c

```diff
@@ -0,0 +1,114 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <kmyth/kmyth.h>
+#include <kmyth/file_io.h>
+
+#include "util.h"
+#include "pelz_log.h"
+#include "pelz_loaders.h"
+#include "pelz_request_handler.h"
+
+#include "sgx_urts.h"
+#include "sgx_seal_unseal_impl.h"
+#include "pelz_enclave.h"
+#include "pelz_enclave_u.h"
+
+int pelz_load_key_from_file(char *filename, charbuf * key)
+{
+  size_t key_len;
+
+  unsigned char tmp_key[MAX_KEY_LEN];
+  FILE *key_file_handle = NULL;
+
+  key_file_handle = fopen(filename, "r");
+  if (key_file_handle == NULL)
+  {
+    pelz_log(LOG_ERR, "Failed to open key file %s", filename);
+    return 1;
+  }
+
+  key_len = fread(tmp_key, sizeof(char), MAX_KEY_LEN, key_file_handle);
+
+  // If we've not reached EOF something has probably gone wrong.
+  if ((key_len == 0) || (!feof(key_file_handle)))
+  {
+    pelz_log(LOG_ERR, "Error: Failed to fully read key file.");
+    secure_memset(tmp_key, 0, key_len);
+    fclose(key_file_handle);
+    return 1;
+  }
+  fclose(key_file_handle);
+
+  *key = new_charbuf(key_len);
+  if (key->len == 0)
+  {
+    pelz_log(LOG_ERR, "Error: Failed to allocate memory for key.");
+    return 1;
+  }
+  memcpy(key->chars, tmp_key, key->len);
+  secure_memset(tmp_key, 0, key_len);
+  return 0;
+}
+
+int pelz_load_file_to_enclave(char *filename, uint64_t * handle)
+{
+  int ext;
+  uint8_t *data = NULL;
+  size_t data_len = 0;
+  uint8_t *data_out = NULL;
+  size_t data_out_len = 0;
+
+  if (read_bytes_from_file(filename, &data, &data_len))
+  {
+    pelz_log(LOG_ERR, "Unable to read file %s ... exiting", filename);
+    return (1);
+  }
+  pelz_log(LOG_DEBUG, "Read %d bytes from file %s", data_len, filename);
+
+  ext = get_file_ext(filename);
```

Comment:
Assuming we don't make this function handle non-```nkl``` or ```ski``` loading we should move the file extension check to before the read, and only attempt the read if the extension isn't ```NO_EXT```.